src/som/compiler/LexicalBuilder.java

```diff
@@ -0,0 +1,82 @@
+package som.compiler;
+
+import som.interpreter.LexicalScope;
+
+public class LexicalBuilder {
+
+  private final LexicalBuilder outerBuilder;
+  private int                  levelsToNextMixin = -1;
+  private int                  levelsToNextMethod = -1;
+
+  public LexicalBuilder(final LexicalBuilder outerBuilder) {
+    this.outerBuilder = outerBuilder;
+  }
+
+  public LexicalScope getScope() {
+    return null;
+  }
+
+  public LexicalBuilder getContextualSelf() {
+    return outerBuilder;
+  }
+
+  private MixinBuilder bubbleToNextMixinOrNull() {
```

Comment:
can we find a nicer word for `bubble`? Is 'outer' to overloaded? 
But it seems to be like outer could be the right word here?